tests/unit/aiplatform/test_jobs.py

```diff
@@ -988,48 +1037,22 @@ def get_mdm_job_mock():
     with mock.patch.object(
         _TEST_API_CLIENT, "get_model_deployment_monitoring_job"
     ) as get_mdm_job_mock:
-        get_mdm_job_mock.return_value = (
-            gca_model_deployment_monitoring_job_compat.ModelDeploymentMonitoringJob(
-                name=_TEST_MDM_JOB_NAME,
-                display_name=_TEST_DISPLAY_NAME,
-                state=_TEST_JOB_STATE_RUNNING,
-                endpoint=_TEST_ENDPOINT,
-            )
-        )
+        get_mdm_job_mock.side_effect = [
+            _TEST_MDM_OLD_JOB,
+            _TEST_MDM_OLD_JOB,
+            _TEST_MDM_OLD_JOB,
+            _TEST_MDM_EXPECTED_NEW_JOB,
+            _TEST_MDM_EXPECTED_NEW_JOB,
+        ]
```

Comment:
This test fixture mocks out the states of the MDM job during the execution of `test_update_mdm_job`. The pytest mock of `get_model_deployment_monitoring_job` is called in the following order:
1. at the beginning of the test definition, `job = jobs.ModelDeploymentMonitoringJob(...)`, during init call of underlying base job class
2.  at the beginning of the test definition, `job = jobs.ModelDeploymentMonitoringJob(...)`, during init call of the `ModelDeploymentMonitoringJob` class
3. at the beginning of `MDMJob.update()`, during `self._sync_gca_resource()`
4. at the beginning of `MDMjob.update()`, during `current_job = self.api_client.get_model_deployment_monitoring_job`
5. at the end of `MDMJob.update()`, during re-assignment of `self._gca_resource`